R/archive.R

```diff
@@ -196,572 +187,755 @@ next_after.Date <- function(x) x + 1L
+  }
 
-        # Keeps LOCF values, such as to be printed
-        keep_locf <- function(df) {
-          dplyr::filter(df, if_all(c(everything(), -version), ~ is_locf(.)))
-        }
+  # Keeps LOCF values, such as to be printed
+  keep_locf <- function(df) {
+    dplyr::filter(df, if_all(c(everything(), -version), ~ is_locf(.)))
+  }
 
-        # Runs compactify on data frame
-        if (is.null(compactify) || compactify == TRUE) {
-          elim <- keep_locf(DT)
-          DT <- rm_locf(DT) # nolint: object_name_linter
-        } else {
-          # Create empty data frame for nrow(elim) to be 0
-          elim <- tibble::tibble()
-        }
+  # Runs compactify on data frame
+  if (is.null(compactify) || compactify == TRUE) {
+    elim <- keep_locf(DT)
+    DT <- rm_locf(DT) # nolint: object_name_linter
+  } else {
+    # Create empty data frame for nrow(elim) to be 0
+    elim <- tibble::tibble()
+  }
 
-        # Warns about redundant rows
-        if (is.null(compactify) && nrow(elim) > 0) {
-          warning_intro <- cli::format_inline(
-            "Found rows that appear redundant based on
+  # Warns about redundant rows
+  if (is.null(compactify) && nrow(elim) > 0) {
+    warning_intro <- cli::format_inline(
+      "Found rows that appear redundant based on
             last (version of each) observation carried forward;
             these rows have been removed to 'compactify' and save space:",
-            keep_whitespace = FALSE
-          )
-          warning_data <- paste(collapse = "\n", capture.output(print(elim, topn = 3L, nrows = 7L)))
-          warning_outro <- cli::format_inline(
-            "Built-in `epi_archive` functionality should be unaffected,
+      keep_whitespace = FALSE
+    )
+    warning_data <- paste(collapse = "\n", capture.output(print(elim, topn = 3L, nrows = 7L)))
+    warning_outro <- cli::format_inline(
+      "Built-in `epi_archive` functionality should be unaffected,
             but results may change if you work directly with its fields (such as `DT`).
             See `?as_epi_archive` for details.
             To silence this warning but keep compactification,
             you can pass `compactify=TRUE` when constructing the archive.",
-            keep_whitespace = FALSE
-          )
-          warning_message <- paste(sep = "\n", warning_intro, warning_data, warning_outro)
-          rlang::warn(warning_message, class = "epiprocess__compactify_default_removed_rows")
-        }
+      keep_whitespace = FALSE
+    )
+    warning_message <- paste(sep = "\n", warning_intro, warning_data, warning_outro)
+    rlang::warn(warning_message, class = "epiprocess__compactify_default_removed_rows")
+  }
+
+  structure(
+    list(
+      DT = DT,
+      geo_type = geo_type,
+      time_type = time_type,
+      additional_metadata = additional_metadata,
+      clobberable_versions_start = clobberable_versions_start,
+      versions_end = versions_end,
+      private = list()
```

Comment:
thought: not sure we need `private` right now since it's empty, unless you're sure this is the way to go & want to set up `print` for it ahead of time.

Reply:
Reading through the code now, I see this isn't used. We do have a `private` field in grouped_epi_archive and that _is_ used, so I'll keep that one, but remove this one.